python/insights_ansible_playbook_lib/_keygen.py

```diff
@@ -0,0 +1,214 @@
+            f"{instructions_file}",
+        ]
+    )
+
+    return gpg_tmp_dir
+
+
+def _export_key_pair(gpg_tmp_dir: str, keys_path: str) -> None:
+    """
+    Export the generated key pair to the `keys_path` directory.
+
+    :param gpg_tmp_dir: The GPG home directory where the key pair was generated.
+    :param keys_path: The directory where the key pair should be exported.
+    """
+    _run_gpg_command(
+        [
+            "/usr/bin/gpg",
+            "--homedir",
+            gpg_tmp_dir,
+            "--export",
+            "--armor",
+            "--yes",
+            "--output",
+            f"{keys_path}/key.public.gpg",
+        ]
+    )
+
+    _run_gpg_command(
+        [
+            "/usr/bin/gpg",
+            "--homedir",
+            gpg_tmp_dir,
+            "--export-secret-keys",
+            "--armor",
+            "--yes",
+            "--output",
+            f"{keys_path}/key.private.gpg",
+        ]
+    )
+
+
+def _get_fingerprint(gpg_tmp_dir: str, keys_path: str) -> str:
+    """
+    Get the fingerprint of the generated key pair and write it to a file in the `keys_path` directory.
+
+    :param gpg_tmp_dir: The GPG home directory where the key pair was generated.
+    :param keys_path: The directory where the key pair is exported.
+    """
+    result = _run_gpg_command(
+        [
+            "/usr/bin/gpg",
+            "--homedir",
+            gpg_tmp_dir,
+            "--fingerprint",
+            "insights-ansible-playbook-verifier test",
+        ],
+        fingerprint=True,
+    )
+
+    # Extract the fingerprint from the output
+    match = re.search(r"^\s+([A-F0-9\s]+)", result.stdout, re.MULTILINE)
+    gpg_fingerprint = match.group(1).strip() if match else ""
+
+    # Write the fingerprint to a file
+    with open(f"{keys_path}/fingerprint.txt", "w") as fingerprint_file:
+        fingerprint_file.write(gpg_fingerprint)
+        logger.debug(
+            f"GPG fingerprint written to a file {fingerprint_file.name}: {gpg_fingerprint}"
+        )
+
+    return gpg_fingerprint
+
+
+def run() -> None:
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--debug",
+        action="store_true",
+        help="Display logs",
+    )
+    parser.add_argument(
+        "--keys",
+        type=pathlib.Path,
+        default=pathlib.Path.cwd(),
+        help="Directory to store the key pair (default: current working directory)",
+    )
+    args = parser.parse_args()
+
+    # Generate the keys
+    gpg_tmp_dir = _generate_keys()
+
+    # Export the keys to the specified directory
+    os.makedirs(args.keys, exist_ok=True)
+    _export_key_pair(gpg_tmp_dir, str(args.keys))
+
+    # Get the fingerprint of the generated key pair
+    _get_fingerprint(gpg_tmp_dir, str(args.keys))
+
+    # Clean up the temporary directory
+    shutil.rmtree(gpg_tmp_dir)
+
+
+def main() -> None:
+    debug: bool = "--debug" in sys.argv
+    lib._configure_logging(debug=debug)
+
+    try:
+        run()
```

Comment:
We should probably print at least some output when the keys get generated. I propose something like `GPG keys were generated to 'key.public.gpg', 'key.private.gpg', 'fingerprint.txt'.`

Reply:
Added print() message there